src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -370,6 +371,40 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                     }
                 }
             }
+            else if (AdvSimd.IsSupported)
+            {
+                if (offset < (nuint)(uint)length)
+                {
+                    lengthToExamine = GetByteVector128SpanLength(offset, length);
+
+                    Vector128<byte> mask = Vector128.Create((byte)0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
+                    Vector128<byte> values = Vector128.Create(value);
+                    while (lengthToExamine > offset)
+                    {
+                        Vector128<byte> search = LoadVector128(ref searchSpace, offset);
+
+                        Vector128<byte> compareResult = AdvSimd.CompareEqual(values, search);
+                        if (AdvSimd.Arm64.MaxAcross(compareResult).ToScalar() == 0)
+                        {
+                            // Zero flags set so no matches
+                            offset += (nuint)Vector128<byte>.Count;
```

Comment:
Given the above `(nuint)(uint)value`, should this be the same?

Reply:
My comment is same as above. I tried to keep it consistent with other implementations. May be we should revisit to address your comments.